packages/basic/lib/utils.js

```diff
@@ -75,3 +77,80 @@ export function cleanBasePath (basePath) {
 export function ensureTrailingSlash (basePath) {
   return basePath ? `${basePath}${basePath.endsWith('/') ? '' : '/'}` : '/'
 }
+
+// Setup the logger
+export function buildPinoOptions (loggerConfig, serverConfig, serviceId, workerId, serviceOptions, root) {
+  const pinoOptions = {
+    level: loggerConfig?.level ?? serverConfig?.level ?? 'trace'
+  }
+
+  if (serviceId) {
+    pinoOptions.name = serviceId
+  }
+
+  if (typeof serviceOptions.context.worker?.index !== 'undefined') {
+    pinoOptions.base = { pid: process.pid, hostname: hostname(), worker: workerId }
+  }
+
+  let require
+  if (loggerConfig?.formatters) {
+    pinoOptions.formatters = {}
+    if (loggerConfig.formatters.bindings) {
+      require = createRequire(root)
+      // pinoOptions.formatters.bindings = loadModule(require, loggerConfig.formatters.bindings.path)
+      pinoOptions.formatters.bindings = loadFunction(require, loggerConfig.formatters.bindings.path)
+    }
+    if (loggerConfig.formatters.level) {
+      require ??= createRequire(root)
+      // pinoOptions.formatters.level = loadModule(require, loggerConfig.formatters.level.path)
+      pinoOptions.formatters.level = loadFunction(require, loggerConfig.formatters.level.path)
+    }
+  }
+
+  if (loggerConfig?.timestamp !== undefined) {
+    if (typeof loggerConfig.timestamp === 'string') {
+      pinoOptions.timestamp = stdTimeFunctions[loggerConfig.timestamp]
+    } else if (loggerConfig.timestamp.path) {
+      require ??= createRequire(root)
+      // pinoOptions.timestamp = loadModule(require, loggerConfig.timestamp.path)
+      pinoOptions.timestamp = loadFunction(require, loggerConfig.timestamp.path)
+    } else { // boolean
+      pinoOptions.timestamp = loggerConfig.timestamp
+    }
+  }
+
+  if (loggerConfig?.redact) {
+    if (Array.isArray(loggerConfig.redact)) {
+      pinoOptions.redact = loggerConfig.redact
+    } else if (typeof loggerConfig.redact === 'object') {
+      pinoOptions.redact = {
+        paths: loggerConfig.redact.paths,
+        censor: loggerConfig.redact.censor
+      }
+    }
+  }
+
+  return pinoOptions
+}
+
+export function loadFunction (require, file) {
+  try {
+    // Check if the file is a valid path
+    const resolvedPath = require.resolve(file)
+
+    // Load the module
+    const mod = require(resolvedPath)
+
+    // Return the function, which could be the default export or the module itself
+    return typeof mod === 'function' ? mod : mod.default || mod
+  } catch (error) {
+    throw new Error(`Failed to load function from ${file}: ${error.message}`)
+  }
+}
+
+const stdTimeFunctions = {
+  epochTime: pino.stdTimeFunctions.epochTime,
+  unixTime: pino.stdTimeFunctions.unixTime,
+  nullTime: pino.stdTimeFunctions.nullTime,
+  isoTime: pino.stdTimeFunctions.isoTime
+}
```

Comment:
You probably want to move this module to the utils module and reuse in runtime and service, too.